flink-core/src/main/java/org/apache/flink/util/InstantiationUtil.java

```diff
@@ -618,6 +620,18 @@ public static void writeObjectToConfig(Object o, Configuration config, String ke
         }
     }
 
+    public static <T> T uncompressAndDeserializeObject(byte[] bytes, ClassLoader cl)
+            throws IOException, ClassNotFoundException {
+        return uncompressAndDeserializeObject(bytes, cl, false);
+    }
+
+    public static <T> T uncompressAndDeserializeObject(
+            byte[] bytes, ClassLoader cl, boolean isFailureTolerant)
+            throws IOException, ClassNotFoundException {
+        return deserializeObject(
+                new InflaterInputStream(new ByteArrayInputStream(bytes)), cl, isFailureTolerant);
```

Comment:
Have ever considered other `InflaterInputStream`, e.g. `GZIPInputStream`?

Reply:
It seems that GZIPInputStream compresses data more slowly with higher compression ratio. It suits better for large files that need to be archived. Also it's less popular than InflaterInputStream. We can set up a comparison test between these two.

Reply:
The result of comparison between different compression methods are illustrated at https://issues.apache.org/jira/browse/FLINK-23005?focusedCommentId=17386867&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17386867.

Reply:
Thanks for the updates.
Given that the compression will not be invoked for many times and the compression time is relatively short, I think it's fine to choose the most common one.